### PR TITLE
Refactor Type 2 checks

### PR DESCRIPTION
**Pull request by WardBrian**
Similar to #73 but for the consolidated type 2

### CODE REVIEW

File: pytorch_finufft/checks.py [hunk 1]
```diff
@@ -82,60 +82,23 @@ def check_output_shape(ndim: int, output_shape: Union[int, Tuple[int, ...]]) ->
...
+    if len(points.shape) == 1:
+        points_dim = 1
+    elif len(points.shape) == 2:
+        points_dim = points.shape[0]
+    else:
+        raise ValueError("The points tensor must be 1d or 2d")
 
-    if dimension != targets_dim:
+    if points_dim not in {1, 2, 3}:
+        raise ValueError(f"Points can be at most 3d, got {points.shape} instead")
```
Comment (eickenberg) [inline pytorch_finufft/checks.py:99]: `Got {points_dim} instead`?

File: pytorch_finufft/checks.py [hunk 2]
```diff
@@ -21,31 +21,31 @@ def check_devices(*tensors: torch.Tensor) -> None:
             )
 
 
-def check_dtypes(values: torch.Tensor, points: torch.Tensor) -> None:
+def check_dtypes(data: torch.Tensor, points: torch.Tensor, name: str) -> None:
```
Comment (eickenberg) [inline pytorch_finufft/checks.py:24]: `name` variable for better error message formatting, correct?
Possible to put a default value for it? E.g. "first argument" or "input array"?

**Review by eickenberg — APPROVED** [review]
nice, that looks great

File: pytorch_finufft/checks.py (hunk 2 above)
Comment (WardBrian) [reply pytorch_finufft/checks.py:24]: Correct. It is either `Values` or `Targets` for our purposes. I suppose in type 3 maybe it has a third name

**Pull request merged**